src/resistance.c

```diff
@@ -32,160 +35,214 @@ is_within_resistance_range(struct edges view, struct edges target,
+	wl_list_for_each(output, &view->server->outputs, link) {
+		if (!output_is_usable(output)) {
+			continue;
 		}
 
-		if (flags.top == 1) {
-			*y = other_edges.top + border.top;
-		} else if (flags.bottom == 1) {
-			*y = other_edges.bottom - eff_height - border.bottom;
+		struct wlr_box mgeom =
+			output_usable_area_in_layout_coords(output);
+
+		struct wlr_box ol;
+		if (!wlr_box_intersection(&ol, &view->current, &mgeom) &&
+				!wlr_box_intersection(&ol, &new_geom, &mgeom)) {
+			continue;
 		}
 
-		/* reset the flags */
-		flags.left = 0;
-		flags.top = 0;
-		flags.right = 0;
-		flags.bottom = 0;
+		struct edges screen_edges = {
+			.left = mgeom.x,
+			.top = mgeom.y,
+			.right = mgeom.x + mgeom.width,
+			.bottom = mgeom.y + mgeom.height,
+		};
+
+		update_nearest_edge(view_edges, target_edges,
+			screen_edges, rc.screen_edge_strength, next_edges);
 	}
 }
 
 void
-resistance_resize_apply(struct view *view, struct wlr_box *new_view_geo)
+resistance_move_apply(struct view *view, double *x, double *y)
 {
-	struct server *server = view->server;
-	struct output *output;
-	struct wlr_box mgeom, intersection;
-	struct wlr_box vgeom = view->current;
-	struct wlr_box tgeom = *new_view_geo;
+	assert(view);
+
 	struct border border = ssd_get_margin(view->ssd);
-	struct edges view_edges; /* The edges of the current view */
-	struct edges target_edges; /* The desired edges */
-	struct edges other_edges; /* The edges of the monitor/other view */
-	struct edges flags = { 0 };
 
-	view_edges.left = vgeom.x - border.left;
-	view_edges.top = vgeom.y - border.top;
-	view_edges.right = vgeom.x + vgeom.width + border.right;
-	view_edges.bottom = vgeom.y + vgeom.height + border.bottom;
+	struct edges next_edges = {
+		.left = INT_MIN,
+		.top = INT_MIN,
+		.right = INT_MAX,
+		.bottom = INT_MAX,
+	};
+
+	struct wlr_box new_geom = {
+		.x = *x,
+		.y = *y,
+		.width = view->current.width,
+		.height = view->current.height,
+	};
+
+	find_screen_edges(view, new_geom, &next_edges, /* move */ true);
+	find_neighbor_edges(view, new_geom, &next_edges, /* move */ true);
+
+	if (next_edges.left > INT_MIN) {
+		*x = next_edges.left + border.left;
+	} else if (next_edges.right < INT_MAX) {
+		*x = next_edges.right - view->current.width - border.right;
+	}
+
+	if (next_edges.top > INT_MIN) {
+		*y = next_edges.top + border.top;
+	} else if (next_edges.bottom < INT_MAX) {
+		*y = next_edges.bottom - border.bottom
+			- view_effective_height(view, /* use_pending */ false);
+	}
+}
 
-	target_edges.left = new_view_geo->x - border.left;
-	target_edges.top = new_view_geo->y - border.top;
-	target_edges.right = new_view_geo->x + new_view_geo->width
-		+ border.right;
-	target_edges.bottom = new_view_geo->y + new_view_geo->height
-		+ border.bottom;
+void
+resistance_resize_apply(struct view *view, struct wlr_box *new_geom)
+{
+	assert(view);
 
-	if (!rc.screen_edge_strength) {
+	/* Shaded windows shoudln't be resized, so ignore resistance */
+	if (view->shaded) {
```

Comment:
Will `view->shaded` ever happen here? If so, should we carry out that check in the calling function?

Reply:
This could probably be dropped altogether or, if retained, converted to an assert. I don't think this function will ever be entered with a shaded view.